### PR TITLE
DS-3279: Retrieval of Anonymous group's UUID for read rights query

### PR DESCRIPTION
**Pull request by jonas-atmire**
PR for jira ticket
https://jira.duraspace.org/browse/DS-3279

Retrieve UUID from anonymous group

### CODE REVIEW

**Comment by terrywbrady** [comment]
@jonas-atmire , could a similar issue exist for permissions in the OAI repository?

**Comment by tdonohue** [comment]
+1 via code review. The code looks correct and this looks like an obvious bug to me.  Needs a quick test though before merging.  

I agree with @terrywbrady that we should double check this same issue doesn't exist in other Solr indexes.

File: dspace-api/src/main/java/org/dspace/discovery/SolrServiceResourceRestrictionPlugin.java [hunk 1]
```diff
@@ -69,7 +69,12 @@ public void additionalSearchParameters(Context context, DiscoverQuery discoveryQ
             if(!authorizeService.isAdmin(context)){
             	StringBuilder resourceQuery = new StringBuilder();
                 //Always add the anonymous group id to the query
-                resourceQuery.append("read:(g0");
+                Group anonymousGroup = groupService.findByName(context,Group.ANONYMOUS);
+                String anonGroupId = "";
+                if(anonymousGroup!=null){
+                    anonGroupId = anonymousGroup.getID().toString();
+                }
+                resourceQuery.append("read:(g"+anonGroupId);
```
Comment (mwoodiupui) [inline dspace-api/src/main/java/org/dspace/discovery/SolrServiceResourceRestrictionPlugin.java]: 'resourceQuery.append("read:(g").append(anonGroupId)' might be a tad more efficient, but this works.

**Comment by aschweer** [comment]
I've checked the oai and statistics cores; neither appears to store group IDs. The statistics core stores the eperson ID if a logged-in user accesses an item/bitstream/etc and this is using the uuid. I don't believe the eperson information is actually ever used by the default stats code anyway. I haven't checked the authority core but the solr schema for that core looks like it doesn't store this type of information at all.

**Comment by aschweer** [comment]
+1 to this PR after testing

**Comment by tdonohue** [comment]
Gave this a test as well today. It works. As I already voted +1 and @aschweer tested/voted as well, I'm going to merge this fix immediately.

**Pull request merged**